src/main/java/com/avail/optimizer/L2Optimizer.java

```diff
@@ -1311,13 +1765,24 @@ private void sanityCheck (final Interpreter interpreter)
 	 */
 	public void optimize (final Interpreter interpreter)
 	{
-		for (final OptimizationPhase phase : OptimizationPhase.values())
+		try
 		{
-			final long before = captureNanos();
-			phase.action.value(this);
-			final long after = captureNanos();
-			phase.stat.record(after - before, interpreter.interpreterIndex);
 			sanityCheck(interpreter);
+			for (final OptimizationPhase phase : OptimizationPhase.values())
+			{
+				final long before = captureNanos();
+				phase.action.value(this);
+				final long after = captureNanos();
+				phase.stat.record(after - before, interpreter.interpreterIndex);
+				sanityCheck(interpreter);
+			}
```

Comment:
it's very elegant how you've done the optimization phases. I like this a lot.

Reply:
It also makes it convenient to put a breakpoint on phase.action.value(this), so I can figure out which phase breaks something.